### PR TITLE
https://github.com/AdguardTeam/AdguardBrowserExtension/issues/124 saf…

### PR DESCRIPTION
**Pull request by Mizzick**
…ari preload

### CODE REVIEW

File: Extension/lib/filter/request-blocking.js [hunk 1]
```diff
@@ -60,7 +61,9 @@ WebRequestService.prototype.processGetSelectorsAndScripts = function (tab, docum
     var genericHideRule = this.antiBannerService.getRequestFilter().findWhiteListRule(documentUrl, documentUrl, "GENERICHIDE");
     var elemHideRule = this.antiBannerService.getRequestFilter().findWhiteListRule(documentUrl, documentUrl, "ELEMHIDE");
     if (!elemHideRule) {
-        if ((Utils.isFirefoxBrowser() && userSettings.collectHitsCount()) || Utils.isContentBlockerEnabled()) {
+        console.warn("Loading all selectors:" + loadAllSelectors);
```
Comment (atropnikov) [inline Extension/lib/filter/request-blocking.js]: Remove this line. console is undefined in Firefox.

File: Extension/lib/content-script/preload.js [hunk 2]
```diff
@@ -62,10 +62,12 @@ var PreloadHelper = {
...
      */
     tryLoadCssAndScripts: function () {
+        var self = this;
         contentPage.sendMessage(
             {
                 type: 'getSelectorsAndScripts',
-                documentUrl: window.location.href
+                documentUrl: window.location.href,
+                loadAllSelectors: self.collapseAllElements
             },
```
Comment (atropnikov) [inline Extension/lib/content-script/preload.js]: Use this instead of self.

Comment (Mizzick) [inline Extension/lib/content-script/preload.js]: nope, it's in an other scape

Comment (atropnikov) [inline Extension/lib/content-script/preload.js]: No, you are in right scope.

File: Extension/lib/content-script/preload.js [hunk 3]
```diff
@@ -62,10 +62,12 @@ var PreloadHelper = {
      * Loads CSS and JS injections
      */
     tryLoadCssAndScripts: function () {
+        var self = this;
         contentPage.sendMessage(
             {
                 type: 'getSelectorsAndScripts',
-                documentUrl: window.location.href
+                documentUrl: window.location.href,
+                loadAllSelectors: self.collapseAllElements
```
Comment (ameshkov) [inline Extension/lib/content-script/preload.js]: @Mizzick so? Why not "this"? This is clearly the same scope.

File: Extension/lib/filter/request-blocking.js [hunk 4]
```diff
@@ -36,11 +36,12 @@ var WebRequestService = exports.WebRequestService = function (framesMap, antiBan
 /**
  * Prepares CSS and JS which should be injected to the page.
  *
- * @param tab           Tab
- * @param documentUrl   Document URL
+ * @param tab                   Tab
+ * @param documentUrl           Document URL
+ * @param loadAllSelectors      Collapse all elements
```
Comment (ameshkov) [inline Extension/lib/filter/request-blocking.js]: Would be really great if you add a link to github issue here.

File: Extension/lib/content-script/preload.js (hunk 3 above)
Comment (Mizzick) [inline Extension/lib/content-script/preload.js]: Because "this" doesn't work?!

Comment (ameshkov) [inline Extension/lib/content-script/preload.js]: But that's simply impossible...

File: Extension/lib/content-script/preload.js [hunk 5]
```diff
@@ -89,7 +89,7 @@ var PreloadHelper = {
         } else {
             this._applySelectors(response.selectors);
             this._applyScripts(response.scripts);
-            if (this.collapseAllElements) {
+            if (this.collapseAllElements || response.allSelectorsLoaded === true) {
```
Comment (ameshkov) [inline Extension/lib/content-script/preload.js]: At least let's rename it to show that the only place where we need it is Safari 9+

File: Extension/lib/filter/request-blocking.js [hunk 6]
```diff
@@ -54,16 +54,24 @@ WebRequestService.prototype.processGetSelectorsAndScripts = function (tab, docum
         return null;
     }
 
+    var loadAllSelectors = !Utils.isContentBlockerEnabled() && !(Utils.isFirefoxBrowser() && userSettings.collectHitsCount());
```
Comment (ameshkov) [inline Extension/lib/filter/request-blocking.js]: Lets move 57-63 to a function

File: Extension/lib/content-script/preload.js (hunk 5 above)
Comment (Mizzick) [inline Extension/lib/content-script/preload.js]: This is not true, all selectors could be loaded in FF also

Comment (ameshkov) [inline Extension/lib/content-script/preload.js]: Why? collapseAllElements is enough

**Pull request merged**